sourced/ml/algorithms/token_parser.py

```diff
@@ -1,29 +1,50 @@
+from enum import Enum
 import re
 
 import Stemmer
 
 
+class TokenMeta(Enum):
```

Comment:
More looks like `TokenStyle` if we call part of the identifier as Token. Or even `SubtokenStyle` to not confuse with Identifier as a code token.

Reply:
\+1 for TokenStyle